pkg/apis/eventing/v1alpha1/trigger_validation.go

```diff
@@ -21,50 +21,48 @@ import (
 	"strconv"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"knative.dev/eventing-rabbitmq/pkg/utils"
 	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
-	"knative.dev/eventing/pkg/client/clientset/versioned"
 	"knative.dev/eventing/pkg/client/injection/client"
 	"knative.dev/pkg/apis"
 	"knative.dev/pkg/kmp"
+	"knative.dev/pkg/webhook/resourcesemantics"
 )
 
 const (
 	BrokerClass           = "RabbitMQBroker"
 	parallelismAnnotation = "rabbitmq.eventing.knative.dev/parallelism"
 )
 
-func ValidateTrigger(ctx context.Context) func(context.Context, *unstructured.Unstructured) error {
-	c := client.Get(ctx)
+// +k8s:controller-gen=false
+type RabbitTrigger struct {
+	metav1.TypeMeta `json:",inline"`
 
-	return func(ctx context.Context, unstructured *unstructured.Unstructured) error {
-		return validate(ctx, unstructured, &RabbitTrigger{Client: c})
-	}
-}
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-// +k8s:deepcopy-gen=false
-type RabbitTrigger struct {
-	eventingv1.Trigger
-	Client versioned.Interface
+	Spec eventingv1.TriggerSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status eventingv1.TriggerStatus `json:"status,omitempty"`
```

Comment:
Why is the status optional now?

Reply:
This is just a direct copy of the eventing [core api](https://github.com/knative/eventing/blob/main/pkg/apis/eventing/v1/trigger_types.go#L45-L57). I think status has to be optional since it's only set once the controller reconciles the resource.